### PR TITLE
Feature/06 app setup

### PR DESCRIPTION
**Pull request by KlugeCh**
@taitruong @RouvenGonzalez 
Please review the basic setup for the website app.
Basic App setup needs to be merged to develop branch and later in main branch to continue with deployment.

basic setup containing:

- installed express.js
- installed and configured ESLint
- installed and configured Prettier
- installed and configured eslint-config-prettier
- configured .gitignore
- installed and configured Jest for unit tests
- installed and configured Playwright for e2e tests
- installed eslint-plugin-jest-playwright
- installed jest-playwright-preset
- added README with description and instructions
- installed express-generator

For testing / running the app local, please check the README.md

### CODE REVIEW

**Review by RouvenGonzalez — APPROVED** [review]
Sieht gut aus so als Basis :+1: 
Eventuell brauchen wir zu nem späteren Zeitpunkt eine andere Ordnerstrucktur als das was der Generator erstellt hat, aber darum können wir uns kümmern wenn es soweit ist.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "commonjs": true,
+    "es2021": true,
+    "jest": true
+  },
+  "extends": ["eslint:recommended", "plugin:jest-playwright/recommended", "prettier"],
+  "parserOptions": {
+    "ecmaVersion": 12
+  },
+  "rules": {
+    "indent": ["error", "tab"],
```
Comment (taitruong) [inline .eslintrc.json]: Please check and test: rule should be: 2 spaces (and no tab)

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "eslint.format.enable": true,
+    "editor.formatOnSave": true,
+    "[javascript]": {
+      "editor.formatOnSave": true
+    },
+    "editor.codeActionsOnSave": {
+      "source.fixAll.eslint": true
+    },
+    "eslint.alwaysShowStatus": true,
+    "files.autoSave": "onFocusChange"
```
Comment (taitruong) [inline .vscode/settings.json:13]: What does that mean?

File: app.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+var createError = require('http-errors');
```
Comment (taitruong) [inline app.js:1]: TODO: next story: convert to typescript, no JS

File: package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "website",
+  "version": "1.0.0",
```
Comment (taitruong) [inline package.json]: semantic versioning: "1.0.0" means this is an official release. Use another version.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  "scripts": {
+    "start": "node ./bin/www",
+    "test": "jest",
+    "e2e-test": "jest --config jest.e2e.config.js",
+    "lint": "eslint .",
+    "fix": "eslint --fix ."
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/software-developer-org/website.git"
```
Comment (taitruong) [inline package.json]: What is that? This URL is unknown

File: package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  "author": "Tai Truong Consulting",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/software-developer-org/website/issues"
+  },
+  "dependencies": {
+    "cookie-parser": "~1.4.4",
+    "debug": "~2.6.9",
+    "eslint": "^7.19.0",
```
Comment (taitruong) [inline package.json:30]: Some packages should be defined as devDependencies

File: public/stylesheets/style.css [hunk 7]
```diff
@@ -0,0 +1,8 @@
+body {
```
Comment (taitruong) [inline public/stylesheets/style.css:1]: CSS and not SCSS?

File: routes/index.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
```
Comment (taitruong) [inline routes/index.js:1]: Next story: typescript

File: README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Run website you can:
+
+```bash
+# either start ...
+npm start
+# ... or start in debug mode
+DEBUG=website:* npm start
+# now open localhost:3000 in your browser
+```
```
Comment (taitruong) [inline README.md:21]: A chapter for folder structure would be good. Also maybe some resources to express. Also maybe a simple explanation about the key files (e.g. templates).

File: .eslintrc.json [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{
+  "env": {
```
Comment (RouvenGonzalez) [inline .eslintrc.json:2]: test kommentar

Comment (taitruong) [reply .eslintrc.json:2]: Ich kann es sehen

File: .vscode/settings.json (hunk 2 above)
Comment (KlugeCh) [reply .vscode/settings.json:13]: - Tested:
Current file is auto saved by changing to another file.
Can be deleted if this behavior is not wanted.
- Description:
https://code.visualstudio.com/docs/editor/codebasics#_save-auto-save

File: .eslintrc.json (hunk 1 above)
Comment (KlugeCh) [reply .eslintrc.json]: According to https://eslint.org/docs/rules/indent#rule-details, ESLint shows an 'error' if indentation is not equal to 'tab',
'tab' default style is 4 spaces.
Fixed, changed to "indent": ["error", 2], for 2 space indentation.

File: package.json (hunk 5 above)
Comment (KlugeCh) [reply package.json]: URL corrected and tested, know working

File: package.json (hunk 6 above)
Comment (KlugeCh) [reply package.json:30]: Splitted in dependencies and devDependencies.
 
"dependencies" are set by default through express-generator 
-   "cookie-parser"
 -   "debug"
 -   "express"
 -   "http-errors"
 -   "morgan"
 -   "pug"
  
"devDependencies":
 -   "eslint"
 -   "eslint-config-prettier"
 -   "eslint-plugin-jest-playwright"
 -   "jest"
 -   "jest-playwright-preset"
 -   "nodemon"
 -   "playwright"
 -   "prettier"

File: public/stylesheets/style.css (hunk 7 above)
Comment (KlugeCh) [reply public/stylesheets/style.css:1]: CSS file was auto generated by express-generator.
Changing to SCSS is part of https://github.com/software-developer-org/website/issues/8

File: .eslintrc.json (hunk 1 above)
Comment (KlugeCh) [reply .eslintrc.json]: Corrected rule in .prettierrc, indentation is now 2 spaces.

File: package.json (hunk 4 above)
Comment (KlugeCh) [reply package.json]: Changed to 0.0.1-alpha.1 for marking as pre-release, according to semver:
- https://softwareengineering.stackexchange.com/questions/275358/version-number-for-a-pre-release-software-that-will-be-the-second-major-release

File: app.js (hunk 3 above)
Comment (KlugeCh) [reply app.js:1]: Created Issue for converting to typescript.

File: routes/index.js (hunk 8 above)
Comment (KlugeCh) [reply routes/index.js:1]: Created Issue for converting to typescript.

File: package.json [hunk 11]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "website",
+  "version": "0.0.1-alpha.1",
```
Comment (taitruong) [inline package.json:3]: @KlugeCh, ist das die korrekte Syntax bei Sematic Versioning? 0.0.0-alpha.1 habe ich noch nie gesehen.

Comment (KlugeCh) [reply package.json:3]: @taitruong 0.0.1-alpha ist korrekt

**Pull request merged**